### ISSUE TITLE
Weird graphical glitch

### ISSUE THREAD

**Issue by gigamonkey** [jmoenig/Snap#2993]
Today working on an animation fairly frequently my Snap workspace would glitch out and suddenly the animation would start drawing in miniature in the middle of the screen. Window screen cap here:

![image](https://user-images.githubusercontent.com/250053/156090005-e98cc256-a707-4410-9519-4970fad19137.png)

When the mini me window comes in it doesn't just blit in as a rectangle but kind of draws in pieces, almost like it's drawing where the sprites are moving but within the miniature window.

**Comment by gigamonkey** [comment]
Here's the project. It doesn't happen all the time and I haven't been able to figure out how to cause it to happen.

https://snap.berkeley.edu/snap/snap.html#present:Username=gigamonkey&ProjectName=U5L2-Simulation

**Comment by gigamonkey** [comment]
Another screen shot

![image](https://user-images.githubusercontent.com/250053/156108323-41245657-4b43-47c0-b659-21473e86b19c.png)

**Comment by jmoenig** [comment]
Hi, Peter, which browser are you using? I've seen this particular issue happen in recent versions of Firefox on MacOS, and haven't so far been able to find out exactly what the problem is. It seems to *only* happen in Firefox in the combination with recent MacOS releases when using many (offscreen) Canvas elements. Chromium/Chrome based browsers don't seem to have this problem, neither does Safari.

Generally I've found Firefox to be awesome, but recently browser and OS makers seem to have been breaking Canvas a lot (Google/Chrome far more frequently than Mozilla in my own experience). Usually they fix those issues within a couple of weeks, but this particular glitch has been going on in Firefox on MacOS for some months now. It seems to be linked to Snap projects using many "clones", which is using more Canvas instances. My intermediate plan is to switch Snap over to the OffscreenCanvas API, but alas, Firefox isn't yet supporting any of that.

**Comment by jguille2** [comment]
I only comment... that I can't reproduce this issue on Linux (neither Chromium nor Firefox)

**Comment by gigamonkey** [comment]
Yeah, I'm using Firefox on MacOS. And indeed this project is using many clones. I'll try Chrome and/or Safari then. Thanks.
